lib/cloudant.js

```diff
@@ -151,17 +151,27 @@ Cloudant.prototype.create = function(model, data, options, cb) {
   var self = this;
   var idName = self.idName(model);
   var mo = self.selectModel(model);
-  mo.db.insert(self.toDB(model, mo, data), function(err, result) {
-    debug('Cloudant.prototype.create insert %j %j', err, result);
-    if (err) {
-      if (err.statusCode === 409) {
-        err.message = err.message + ' (duplicate?)';
+  if (options.upsert === false) {
+    self.findById(model, data.id, {}, function(err) {
```

Comment:
I find this confusing. Why is `create` method querying the database? What if `data.id` is not set, because it's auto-generated by the database? 

The property name `id` should not be hard-coded.

`Cloudant.prototype.create` is a public method used by DAO method in juggler. I don't think it's a good idea to add `options.support` to a method like this, because now users can do something like this:

```
Product.create({name: 'Pen' }, { upsert: false }, cb);
```

(No other connectors support `options.upsert` in `create` to my best knowledge.)